api-gateway/src/test/java/com/petclinic/bffapigateway/presentationlayer/ApiGatewayControllerTest.java

```diff
@@ -148,7 +160,30 @@ void createUser(){
         assertEquals(user.getUsername(), "Johnny123");
         assertEquals(user.getPassword(), "password");
         assertEquals(user.getEmail(), "email@email.com");
+    }
 
+    @Test
+    void updateUser(){
+        UserDetails user = new UserDetails();
+        //user.setId(1);
+        user.setUsername("Johnny123");
+        user.setPassword("password");
+        user.setEmail("email@email.com");
+        when(authenticationServiceClient.updateUser(1, user)).thenReturn(Mono.just(user));
+
+        client.put()
+                .uri("/api/gateway/users/{userId}", 1)
+                .body(Mono.just(user), UserDetails.class)
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBody();
+
+        //assertEquals(user.getId(),1);
+        assertEquals(user.getUsername(), "Johnny123");
+        assertEquals(user.getPassword(), "password");
+        assertEquals(user.getEmail(), "email@email.com");
```

Comment:
It would be nice if you could assert the response from the API, instead of asserting the property fields.

Reply:
It would be nice if you stay silent

Reply:
I think that Kenny is right, what do you think?

Reply:
You only tested your ok case. No tests for error cases ?